doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,712 @@
+    "meter_type":"packets",
+    "mode":"sr_tcm",
+    "red_action":"drop",
+    "trap_action":"trap",
+    "trap_priority":"7",
+    "queue": "7",
+    "yellow_action":"drop"
+  }
+  ...
+  "vrrp": {
+    "trap_ids": "vrrp",
+    "trap_group": "trap.group.vrrp",
+    "always_enabled": "true"
+  },
+  "vrrpv6": {
+    "trap_ids": "vrrpv6",
+    "trap_group": "trap.group.vrrpv6",
+    "always_enabled": "true"
+  }
+```
+#### DB Changes
+
+##### CONFIG_DB changes
+
+VRRP
+
+Producer: config manager
+
+Consumer: vrrpmgrd and vrrpcfgd/frrcfgd
+
+Description: New table that stores VRRP configuration for per interface + VRID.
+```
+;New table
+;holds the VRRP configuration per interface and VRID
+key             = VRRP:interface_name:vrid
+                                                ; Interface name string. Vlan, Ethernet or PortChannel
+                                                ; vrid is an integer
+; field = value
+vip             = ip_address                    ; Virtual IP address. This is a list of IP addresses
+priority        = 3DIGIT                        ; Priority of VRRP instance
+adv_interval    = 5DIGIT                        ; Advertisement interval for VRRP. Default = 1000ms
+state           = vrrp_state                    ; String denoting the state of VRRP instance
+version         = vrrp_version                  ; VRRP version. Value is 2 or 3. Default is 3
+preempt         = "enabled"/"disabled"          ; VRRP pre-emption is enabled? Default is enabled
+track_interface = interface_Name|weight_value   ;This is repeated for the configured tracking interfaces
+```
+Example:-
+```
+admin@sonic:~$ redis-cli -n 4 keys VRRP*
+
+"VRRP|Ethernet8|8"
+"VRRP|Ethernet1|1"
+"VRRP|Vlan10|10"
+"VRRP|Ethernet2|2"
+"VRRP|Vlan4|4"
+
+admin@sonic:~$ redis-cli -n 4 HGETALL " VRRP|Vlan10|10"
+
+"vrid"
+"10"
+"vip"
+"4.1.1.100/24,4.1.1.200/24,2000::50/64"
+"priority"
+"100"
+"adv_interval"
+"1000"
+"state"
+""
+"version"
+"3"
+"preempt"
+"enabled"
+"track_interface"
+"Ethernet5|10, Ethernet9|20"
+```
+
+#### APPL_DB Changes
+
+VRRP_TABLE
+
+Producer: vrrpmgrd and vrrpsyncd
+
+Consumer: vrrporch
+
+Description: This is a new table that contains VRRP state information. This entry will be added to APP_DB for each VRRP instance.
+```
+; New table
+; holds the VRRP state and VMAC information
+
+key         = VRRP_TABLE:interface_name:vrid:type
+              interface_name        ; interface name as a string. Vlan, Ethernet or PortChannel
+              vrid                  ; vrid is an integer
+              type                  ; IP address type string. IPv4 or IPv6
+    
+; field = value
+vmac        = virtual_mac_address   ; Virtual MAC address associated with VRRP instance
+state       = "up" | "down"         ; Linux macvlan device state
+priority    = vrrp_priority         ; Priority of VRRP instance
+```
+Example:-
+```
+admin@sonic:~$ redis-cli -n 0 HGETALL "VRRP_TABLE:Ethernet36:6:IPv4"
+1) "vmac"
+2) "00:00:5e:00:01:06"
```

Comment:
When you have both vr-id and address family, why cant we derive the vmac in the orchagent internally?

Reply:
good suggestion!!